frontend/src/app/core/services/user.service.ts

```diff
@@ -0,0 +1,15 @@
+import { Injectable } from '@angular/core';
+import { environment } from 'src/environments/environment';
+import { HttpClient } from '@angular/common/http';
+import { User } from '../models/user';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class UserService {
+    constructor(private http: HttpClient) { }
+
+    public updateUser(user: User) {
+        return this.http.put<User>(`${environment.coreUrl}/user`, user);
```

Comment:
Please, use wrapper service (InternalHttpService, but not sure how it's called, try find in core/services) instead of HttpClient